pkgs/applications/networking/irc/url-bot-rs/default.nix

```
@@ -0,0 +1,67 @@
+{ stdenv
+, lib
+, pkgs
+, rustPlatform
+, fetchFromGitHub
+
+# native
+, pkg-config
+
+# build
+, openssl
+, sqlite
+
+# darwin
+, libiconv
+, Security
+}:
+
+let
+  pname = "url-bot-rs";
+  version = "0.3.1";
+  owner = "nuxeh";
+  repo = pname;
+in
+rustPlatform.buildRustPackage {
+  inherit pname version;
+
+  src = fetchFromGitHub {
+    inherit owner repo;
+    rev = "v${version}";
+    sha256 = "0y183z86rkpxxvvzrkkpdxxxpd7lr8nnmwi5nv7k8cfb8fa0xl8n";
+  };
+
+  cargoLock.lockFileContents = builtins.readFile ./Cargo.lock;
```

This should also work:;
```suggestion
  cargoLock.lockFile = ./Cargo.lock;
```

---

You also still have to add the lockfile to the comit.